addons/account/models/partner.py

```diff
@@ -391,6 +391,26 @@ def _get_company_currency(self):
             else:
                 partner.currency_id = self.env.company.currency_id
 
+    @api.model
+    def _default_rank(self, field_name, search_mode):
+        if not search_mode:
+            return 0
+        is_customer = search_mode == 'customer'
+        if is_customer:
+            return 'customer_rank' == field_name
+        is_supplier = search_mode == 'supplier'
+        return is_supplier and 'supplier_rank' == field_name
+
+    @api.model
+    def _default_supplier_rank(self):
+        search_mode = self.env.context.get('res_partner_search_mode')
+        return self._default_rank('supplier_rank', search_mode)
+
+    @api.model
+    def _default_customer_rank(self):
+        search_mode = self.env.context.get('res_partner_search_mode')
+        return self._default_rank('customer_rank', search_mode)
```

Comment:
Is it really necessary all this functions? The parent function is basically 2 separate functions anyway with the early return. Am I missing something or can't these functions literally be like.

 @api.model
    def _default_customer_rank(self):
       return int(self.env.context.get('res_partner_search_mode', '') == 'customer')
```